File: .examples/.dart
class Person {
  String name;
  int age;

  Person(this.name, this.age);

  String greet() {
    return "Hello, my name is $name and I am $age years old.";
  }
}

void main() {
  int currentYear = DateTime.now().year;
  String personName = "Alice";
  int personAge = 30;

  Person person = Person(personName, personAge);
  int yearOfBirth = currentYear - personAge;

  print(person.greet());
  print("${person.name} was born in $yearOfBirth.");
}

File: .examples/.erl
-module(person_example).
-export([main/0, greet/1, calculate_year_of_birth/2]).

-record(person, {name, age}).

greet(#person{name = Name, age = Age}) ->
  io:format("Hello, my name is ~s and I am ~p years old.~n", [Name, Age]).

calculate_year_of_birth(CurrentYear, Age) ->
  CurrentYear - Age.

main() ->
  CurrentYear = calendar:local_time(),
  {Year, _, _} = CurrentYear,
  Name = "Alice",
  Age = 30,
  Person = #person{name = Name, age = Age},
  greet(Person),
  YearOfBirth = calculate_year_of_birth(Year, Age),
  io:format("~s was born in ~p.~n", [Name, YearOfBirth]).
